vhdbuilder/packer/pre-install-dependencies.sh

```diff
@@ -89,4 +89,19 @@ if [[ ${OS} == ${MARINER_OS_NAME} ]] && [[ "${ENABLE_CGROUPV2,,}" == "true" ]];
   enableCgroupV2forAzureLinux
 fi
 
+if [[ "${UBUNTU_RELEASE}" == "22.04" ]]; then
+  echo "Logging the currently running kernel: $(uname -r)"
+  echo "Before purging kernel, here is a list of kernels/headers installed:"; dpkg --list | grep 'linux-image\|linux-headers'
+
+  # Purge all current kernels and dependencies
+  DEBIAN_FRONTEND=noninteractive apt-get remove --purge -y $(dpkg-query -W 'linux-*azure*' | awk '$2 != "" { print $1 }' | paste -s)
+  echo "After purging kernel, dpkg list should be empty"; dpkg --list | grep 'linux-image\|linux-headers'
+
+  # Install lts-22.04 kernel
+  DEBIAN_FRONTEND=noninteractive apt-get install -y linux-image-azure-lts-22.04 linux-cloud-tools-azure-lts-22.04 linux-headers-azure-lts-22.04 linux-modules-extra-azure-lts-22.04 linux-tools-azure-lts-22.04
+  echo "After installing new kernel, here is a list of kernels/headers installed"; dpkg --list | grep 'linux-image\|linux-headers'
+  
+  update-grub
```

Comment:
This shouldn't be necessary; apt has a hook to run update-grub on kernel package changes.

Reply:
discussed offline, this just adds 2/3s to build time, no problem otherwise.